### PR TITLE
Add Retrofit

### PR DESCRIPTION
**Pull request by mdipirro**
I added something about how we communicate with our server.

### CODE REVIEW

File: sections/server.tex [hunk 1]
```diff
@@ -43,3 +43,77 @@ \section{QuizFight server}
...
+	Games.Players.getCurrentPlayer(client).getDisplayName(),
+	token,
+	Secure.getString(getContentResolver(), Secure.ANDROID_ID)
+)).call(new Callback<ResponseBody>() {
+	@Override
+	public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {}
+	
+	@Override
+	public void onFailure(Call<ResponseBody> call, Throwable t) {}
+});
```
Comment (EmanueleC) [inline sections/server.tex]: In a recent issue, this function `call(@NonNull Callback<T> callback)` becomes `call(@NonNull Callback<T> callback, QuizFightApplication application)`, it requires also the application parameter

File: sections/server.tex [hunk 2]
```diff
@@ -43,3 +43,77 @@ \section{QuizFight server}
...
+
+\subsection{Client-server communication}
+For communicating with the server we chose the Retrofit library. It is a 
+type-safe REST client for Android (or just Java) developed by Square. 
+It provides a powerful framework for interacting with APIs and sending 
+network requests with \texttt{OkHttp}.
+
+This library makes downloading JSON or XML data from a web API fairly 
+straightforward. Once the data is downloaded then it is parsed into a 
+Plain Old Java Object (POJO) which must be defined for each ''resource''
```
Comment (EmanueleC) [inline sections/server.tex]: ''resource'' wrong latex quotes (use ``resource'')

File: sections/server.tex [hunk 3]
```diff
@@ -43,3 +43,77 @@ \section{QuizFight server}
...
+straightforward. Once the data is downloaded then it is parsed into a 
+Plain Old Java Object (POJO) which must be defined for each ''resource'' 
+in the response.
+
+The communication obviously requires the \texttt{android.permission.INTERNET}
+permission. We share only JSON objects. As aforementioned, our server is 
+specifically designed for being efficient in JSON handling. Even if a 
+communication paradigm based on XML would be more secure (as XML has a schema
+defining exactly how the document should be), the JSON alternative is simpler 
+and faster. Further, we do not need an absolute confidence in the document. If
```
Comment (EmanueleC) [inline sections/server.tex]: Use "furthermore"

Comment (EmanueleC) [inline sections/server.tex]: Use "Furthermore"

File: sections/server.tex [hunk 4]
```diff
@@ -43,3 +43,77 @@ \section{QuizFight server}
 requests.
 We then prefer using usernames, which we know by Google's assurance to be
 unique.
+
+\subsection{Client-server communication}
+For communicating with the server we chose the Retrofit library. It is a 
+type-safe REST client for Android (or just Java) developed by Square. 
+It provides a powerful framework for interacting with APIs and sending 
+network requests with \texttt{OkHttp}.
```
Comment (EmanueleC) [inline sections/server.tex]: Maybe it's better to specify what OkHttp is (few words, like: OkHttp is an HTTP client, its request/response API is designed with fluent builders and immutability. It supports both synchronous blocking calls and async calls with callbacks.)

**Review by EmanueleC — COMMENTED** [review]
Seems right to me but the connectivity part is missing

File: sections/server.tex (hunk 4 above)
Comment (mdipirro) [reply sections/server.tex]: Done

File: sections/server.tex (hunk 2 above)
Comment (mdipirro) [reply sections/server.tex]: Done

File: sections/server.tex (hunk 3 above)
Comment (mdipirro) [reply sections/server.tex]: Done

File: sections/server.tex (hunk 1 above)
Comment (mdipirro) [reply sections/server.tex]: I modified both the listing and the class diagram. Just add here, if you want, the connectivity part.

**Pull request merged**